services/provider/server/storageclassrequest.go

```diff
@@ -0,0 +1,169 @@
+package server
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+	controllers "github.com/red-hat-storage/ocs-operator/controllers/storageconsumer"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+type storageClassRequestManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newStorageClassRequestManager(ctx context.Context, cl client.Client, namespace string) (*storageClassRequestManager, error) {
+	return &storageClassRequestManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+// getStorageClassRequestName generates a name for a StorageClassRequest resource.
+func getStorageClassRequestName(consumerUUID, storageClassRequestName string) string {
+	var s struct {
+		StorageConsumerUUID     string `json:"storageConsumerUUID"`
+		StorageClassRequestName string `json:"storageClassRequestName"`
+	}
```

Comment:
This can cause migration problems as the json struct is changing and will result in creating new claims. This needs to be taken care of during migration.
cc @nb-ohad cc @rewantsoni